torch/testing/_internal/common_utils.py

```diff
@@ -906,6 +906,9 @@ def assertTensorsEqual(a, b):
         elif isinstance(x, bool) and isinstance(y, bool):
             super(TestCase, self).assertEqual(x, y, message)
         elif isinstance(x, Number) and isinstance(y, Number):
+            # Handle Nans
+            if x != x and y != y:
+                return
```

Comment:
What is the likeliness of this code ever returning?

Reply:
@mmphego It returns when both `x` and `y` are Nans.

Reply:
Thank you for the feedback.